src/Form/FilePersisterServiceSettingsForm.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace Drupal\strawberryfield\Form;
+
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\strawberryfield\Tools\Ocfl\OcflHelper;
+use Drupal\Core\Ajax\InvokeCommand;
+use Drupal\Core\Ajax\MessageCommand;
+
+/**
+ * ConfigurationForm for Strawberryfield File Storage.
+ */
+class FilePersisterServiceSettingsForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'strawberryfield.filepersister_service_settings',
+      'strawberryfield.storage_settings',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'strawberryfield_filepersister_settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('strawberryfield.filepersister_service_settings');
+    $config_storage = $this->config('strawberryfield.storage_settings');
+    $scheme_options = OcflHelper::getVisibleStreamWrappers();
+    $form['file_scheme'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Storage Scheme for Persisting Files'),
+      '#description' => $this->t('Please provide your prefered Storage Scheme for Persisting Strawberryfield managed Files'),
+      '#default_value' => $config_storage ->get('file_scheme'),
+      '#options' => $scheme_options,
+      '#required' => TRUE
+
+    ];
+
+    $form['object_file_scheme'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Storage Scheme for Persisting Digital Objects'),
+      '#description' => $this->t('Please provide your prefered Storage Scheme for Persisting Digital Objects as JSON Files'),
+      '#default_value' => $config_storage ->get('object_file_scheme'),
+      '#options' => $scheme_options,
+      '#required' => TRUE
+    ];
+
+    $form['extractmetadata'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Should File level metadata extraction be processed?'),
+      '#description' => $this->t('If enabled, exiftool and FIDO will run on every file.'),
+      '#default_value' => !empty($config->get('extractmetadata')) ? $config->get('extractmetadata'): FALSE,
+      '#return_value' => TRUE,
+    ];
+    $form['exif_exec_path'] = [
```

Comment:
I need now an extra field here to allow arguments from exiftool, so people can have less EXIF tags extracted. I also want here the discussion point of #86 happening. Basically, don't put EXIF (and same for the flavour version) into the JSON itself, but into a file that we reference.

Reply:
@DiegoPino I fully agree with you to reduce exif output tags.
Exif output into file instead of SBF-JSON, can it make more complex to manage results? It's my only doubt.

Reply:
It can. It also means we need to revive the OCR work we started, since this will/should be indexed the same way. Putting it into a file is simple enough. Getting it back! is another thing. Probably the solution is in between, some in the JSON, others in the file. We need to discuss this for sure

Reply:
> It can. It also means we need to revive the OCR work we started, since this will/should be indexed the same way. Putting it into a file is simple enough. Getting it back! is another thing. Probably the solution is in between, some in the JSON, others in the file. We need to discuss this for sure

@DiegoPino probably exif output is a little bit different from OCR output, I mean regarding output dimensions. For OCR I think JSON+file is ok, for EXIF we have to evaluate if putting result into file generate more CPU cycles to retrieve it than into only JSON

Reply:
I agree. maybe we simply not extract everything and can live with these. I was just getting too much overhead when dealing with EXIF heavy images and a lot of them (and also because of the flattening algorithm... but then now, thinking about this better, i actually can avoid flattening flv: elements!... should i open a ticket??)